From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Fri, 3 May 2019 18:10:47 +0000
Subject: window: Emit an error and return when trying to activate an
 unmanaged

If something (i.e. gnome-shell or an extension) tries to activate an unmanaged
window, we should warn about this and avoid to perform further actions as this
could lead to a crash of mutter, since the window has not valid flags (like
workspace) set anymore at this stage.

(cherry picked from commit a6fc656e917fd48b8708b8d9f4bf9f8b15581313)

Bug: https://gitlab.gnome.org/GNOME/mutter/issues/580
Forwarded: https://gitlab.gnome.org/GNOME/mutter/merge_requests/564
Origin: upstream, 3.30.3, commit:a9322c81af652f50b0c6fcce8e35d5c84948d68a
---
 src/core/window.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/core/window.c b/src/core/window.c
index 78faf38..a584c1f 100644
--- a/src/core/window.c
+++ b/src/core/window.c
@@ -3639,6 +3639,13 @@ meta_window_activate_full (MetaWindow     *window,
 {
   MetaWorkspaceManager *workspace_manager = window->display->workspace_manager;
   gboolean allow_workspace_switch;
+
+  if (window->unmanaging)
+    {
+      g_warning ("Trying to activate unmanaged window '%s'", window->desc);
+      return;
+    }
+
   meta_topic (META_DEBUG_FOCUS,
               "_NET_ACTIVE_WINDOW message sent for %s at time %u "
               "by client type %u.\n",
